### PR TITLE
Prestonc/basex

### PR DESCRIPTION
**Pull request by prestoncarman**
The change moves to Hyracks 0.2.16 and updates the documentation to include diagrams.

### CODE REVIEW

File: vxquery-cli/src/main/java/org/apache/vxquery/cli/VXQuery.java [hunk 1]
```diff
@@ -266,15 +267,16 @@ private void runJob(JobSpecification spec, PrintWriter writer) throws Exception
 
         JobId jobId = hcc.startJob(spec, EnumSet.of(JobFlag.PROFILE_RUNTIME));
 
-        ByteBuffer buffer = ByteBuffer.allocate(spec.getFrameSize());
+        //        ByteBuffer buffer = ByteBuffer.allocate(spec.getFrameSize());
```
Comment (westmann) [inline vxquery-cli/src/main/java/org/apache/vxquery/cli/VXQuery.java]: Can we just remove this line?

File: vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/basex_index/stations_database_remove.bxs [hunk 2]
```diff
@@ -0,0 +1 @@
+DROP DB db_stations
```
Comment (westmann) [inline vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/basex_index/stations_database_remove.bxs]: What are the chars at the end of the file?

**Comment by westmann** [comment]
It seems that I cannot see all the diffs, because too many files changed? 
Is there a way to look at all of them in this WebUI that I haven't found or do I need to check the branch out to see the diff?

**Comment by prestoncarman** [comment]
I switched the pom to Java 1.8. 

Till, I am not sure how to see all the files in the web view. Can you finish your review?

File: src/site/apt/index.apt [hunk 3]
```diff
@@ -22,6 +22,7 @@ Apache VXQuery
 
   There are lots of large collections of relatively small documents like e.g. the {{{http://www.sec.gov/info/edgar/ednews/dissemin.htm}EDGAR dataset}} or the {{{http://wiki.openstreetmap.org/wiki/Download}OpenStreetMap dataset}}.
   However there are no XQuery processors available today that are capable of processing these datasets in parallel and making the contained information accessible.
-  
 
-
```
Comment (sjaco002) [inline src/site/apt/index.apt]: Looks like some weird white space issues on this file. Not sure if it matters or not.

File: vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/basex_scripts/README [hunk 4]
```diff
@@ -0,0 +1,9 @@
+BaseX local configuration options for the weather benchmark.
+The options attempt to match the processing and output of VXQuery.
+
+
+# Local Options
+CHOP = false
+DBPATH = /home/ecarman/disk1/basex/BaseXData
+REPOPATH = /home/ecarman/disk1/basex/BaseXRepo
+WEBPATH = /home/ecarman/disk1/basex/BaseXWeb
```
Comment (sjaco002) [inline vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/basex_scripts/README]: I'm assuming that these overwrite during mvn package? Or is this README a sample guideline?

File: vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/mrql_scripts/run_mrql_tests.sh [hunk 5]
```diff
@@ -40,7 +41,6 @@ if which programname >/dev/null;
 then
     echo "Sending out e-mail notification."
     SUBJECT="MRQL Tests Finished (${DATASET})"
-    EMAIL="ecarm002@ucr.edu"
```
Comment (sjaco002) [inline vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/mrql_scripts/run_mrql_tests.sh:43]: I'm confused here as well. This looks like it is actually sending you an email right?

File: vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/saxon_scripts/run_saxon_tests.sh [hunk 6]
```diff
@@ -35,7 +37,6 @@ if which programname >/dev/null;
 then
     echo "Sending out e-mail notification."
     SUBJECT="Saxon Tests Finished (${DATASET})"
-    EMAIL="ecarm002@ucr.edu"
```
Comment (sjaco002) [inline vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/saxon_scripts/run_saxon_tests.sh:38]: Same as above

File: vxquery-core/src/main/java/org/apache/vxquery/collations/Collation.java [hunk 7]
```diff
@@ -32,4 +32,4 @@
     public CharSequence substringBefore(CharSequence cs1, CharSequence cs2);
 
     public CharSequence substringAfter(CharSequence cs1, CharSequence cs2);
-}
```
Comment (sjaco002) [inline vxquery-core/src/main/java/org/apache/vxquery/collations/Collation.java]: Another weird white space thing

File: vxquery-core/src/main/java/org/apache/vxquery/compiler/algebricks/VXQueryBinaryBooleanInspectorFactory.java [hunk 8]
```diff
@@ -41,4 +41,4 @@ public boolean getBooleanValue(byte[] bytes, int offset, int length) {
             }
         };
     }
-}
```
Comment (sjaco002) [inline vxquery-core/src/main/java/org/apache/vxquery/compiler/algebricks/VXQueryBinaryBooleanInspectorFactory.java]: whitespace

File: vxquery-core/src/main/java/org/apache/vxquery/compiler/algebricks/VXQueryBinaryIntegerInspectorFactory.java [hunk 9]
```diff
@@ -41,4 +41,4 @@ public int getIntegerValue(byte[] bytes, int offset, int length) {
             }
         };
     }
-}
```
Comment (sjaco002) [inline vxquery-core/src/main/java/org/apache/vxquery/compiler/algebricks/VXQueryBinaryIntegerInspectorFactory.java]: whitespace

File: vxquery-core/src/main/java/org/apache/vxquery/compiler/algebricks/VXQueryComparatorFactoryProvider.java [hunk 10]
```diff
@@ -58,4 +58,4 @@ public int compare(byte[] b1, int s1, int l1, byte[] b2, int s2, int l2) {
             };
         }
     }
-}
```
Comment (sjaco002) [inline vxquery-core/src/main/java/org/apache/vxquery/compiler/algebricks/VXQueryComparatorFactoryProvider.java]: whitespace

File: vxquery-core/src/main/java/org/apache/vxquery/compiler/algebricks/VXQueryConstantValue.java [hunk 11]
```diff
@@ -61,4 +61,4 @@ public String toString() {
                 .append("])");
         return buffer.toString();
     }
-}
```
Comment (sjaco002) [inline vxquery-core/src/main/java/org/apache/vxquery/compiler/algebricks/VXQueryConstantValue.java]: whitespace

File: vxquery-core/src/main/java/org/apache/vxquery/compiler/algebricks/VXQueryExpressionRuntimeProvider.java [hunk 12]
```diff
@@ -145,4 +145,4 @@ public IUnnestingEvaluatorFactory createUnnestingFunctionFactory(UnnestingFuncti
             throw new AlgebricksException(e);
         }
     }
-}
```
Comment (sjaco002) [inline vxquery-core/src/main/java/org/apache/vxquery/compiler/algebricks/VXQueryExpressionRuntimeProvider.java]: I'm going to stop commenting this (looks like a lot of files). I'm not sure whether this needs to be changed, but if so, it needs to for all these files.

File: vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/basex_scripts/README (hunk 4 above)
Comment (prestoncarman) [inline vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/basex_scripts/README]: I updated these to be suggested paths (removed my local path). They are for the reader to see and add to their own configuration.

File: vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/mrql_scripts/run_mrql_tests.sh (hunk 5 above)
Comment (prestoncarman) [inline vxquery-benchmark/src/main/resources/noaa-ghcn-daily/other_systems/mrql_scripts/run_mrql_tests.sh:43]: E-mail is now an option to add to the command line.

**Pull request closed without merging**